Adding localization

remil19 opened the pull request:
It's now very simple to translate the web site:
- first create a new file languages/'your lang'.json based on en.json;
- add your language at the first lines of dev/generate-static.php;
- run php generate-static.php in dev/
  a new file for your language should be present at the root.

Any modification at the site should be done at dev/index.php

A small server-side program should determine which language choose for the visitor and redirect to the correct file.

remil19 committed:
Create index.php

remil19 committed:
Create generate-static.php

remil19 committed:
Create en.json

remil19 committed:
Create fr.json

remil19 committed:
French translation generated

remil19 committed:
Update fr.json

Correction typographique

heyimjuani commented:
This is f*cking awesome

> On 02/03/2014, at 04:04, Rémi Lapeyre notifications@github.com wrote:
> 
> It's now very simple to translate the web site:
> 
> first create a new file languages/'your lang'.json based on en.json;
> add your language at the first lines of dev/generate-static.php;
> run php generate-static.php in dev/ a new file for your language should be present at the root.
> Any modification at the site should be done at dev/index.php
> 
> A small server-side program should determine which language choose for the visitor and redirect to the correct file.
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/remil19/popcorn-time.github.io master
> Or view, comment on, or merge it at:
> 
>   https://github.com/popcorn-time/popcorn-time.github.io/pull/6
> 
> Commit Summary
> 
> Create index.php
> Create generate-static.php
> Create en.json
> Create fr.json
> French translation generated
> File Changes
> 
> A dev/generate-static.php (29)
> A dev/index.php (204)
> A fr.html (201)
> M index.html (16)
> A languages/en.json (27)
> A languages/fr.json (27)
> Patch Links:
> 
> https://github.com/popcorn-time/popcorn-time.github.io/pull/6.patch
> https://github.com/popcorn-time/popcorn-time.github.io/pull/6.diff
> —
> Reply to this email directly or view it on GitHub.

remil19 commented:
I've just seen that you're using Github Pages and i think it's only static so maybe you should add flags in order to let the user choose his language or use javascript to redirect to the correct page (remember, any change must be done in dev/index.php, any change in index.html will be over will be overwrited).

fmpaes commented:
+1 for support

remil19 committed:
Correction orthographique

argen commented:
Will help translate to spanish and make some amends for the english version.

argen commented:
@abadfederico can we resolve conflicts and merge this so we can start translating?

c0b41 committed:
turkish language

EvandroZanatta commented:
had cited an example( #17 ) with translated versions of the site, but I believe that using JSON files with the translations would be the best way. Just as it is translated the "App".

remil19 committed:
Merge pull request #1 from cobaimelan/master

turkish language

remil19 committed:
tr.html generated

remil19 commented:
The Turkish a-has just been added. Is there any problem with the content of this pull request because it's been some time and translations could be a plus to the site.

EvandroZanatta commented:
@remil19 I will create a "pt-br.json" and translate ok?

EvandroZanatta committed:
Create a pt-br translate

c0b41 commented:
@remil19  faq page generate translate?

remil19 committed:
Merge pull request #2 from EvandroZanatta/patch-1

Create a pt-br translate

remil19 committed:
Complete the pt-br translation.

remil19 commented:
@cobaimelan there is just few lines to add in order to translate the faq, i will write them soon if i have time to.

@EvandroZanatta Thanks for that i just merged the changes .

EvandroZanatta commented:
I found some errors still exist, I will review.

EvandroZanatta committed:
Spelling and Grammar Review

I have reviewed the translation of every sentence trying to give more sense to the sentences and making the most natual reading. 
Some grammatical errors have also been corrected.

remil19 committed:
Merge pull request #3 from EvandroZanatta/master

Spelling and Grammar Review

remil19 committed:
Generate pt-br.html

remil19 committed:
Few corrections of the brazilian Portuguese translation

c0b41 commented:
@remil19 No wonder it possible to change the font Turkish characters
![adsz](https://f.cloud.github.com/assets/2834954/2386066/eef1dc7e-a926-11e3-9e0b-770cdd5f1b71.png)

remil19 commented:
I've seen that too. I'm not sure that it's the translating engine that should take care of that. Should we change the font for all languages in order to keep a unity or just change the font for the Turkish even if it could be a bit more complicated to maintain?

Moreover when i first forked the project in order to add localization, there was no faq, i can easily add it now but maybe the changes already made should be merged before, that would simpler.

I've noticed that there is an Spanish translation in the main repo, is this pull request still usefull or should i delete it?

remil19 committed:
Create README.md

EvandroZanatta committed:
Creating the "readme" file

Show the "readme" file the steps that must be followed to correctly translate the site.

remil19 committed:
Merge branch 'master' of https://github.com/EvandroZanatta/popcorn-time.github.io into EvandroZanatta-master

remil19 committed:
Update Readme

Merge branch 'master' of github.com:remil19/popcorn-time.github.io

Conflicts:
	README.md

remil19 committed:
Update README.md

comzeradd committed:
Add greek translation

remil19 committed:
Merge branch 'greek' of https://github.com/comzeradd/popcorn-time.github.io into comzeradd-greek

remil19 commented:
@heyimjuani is has been 11 days since this pull request has been opened. There is more differences between the two repositories with time passing. Is the merge between them consider or should we stop to work on the fork?

EvandroZanatta commented:
@remil19 +1

fmpaes commented:
@remil19 +1

Commit:
Update pt-br.html

small fix

Commit:
Update pt-br.json

small fix

remil19 committed:
Merge pull request #7 from felipemario/patch-2

Update pt-br.json

remil19 committed:
Merge pull request #6 from felipemario/patch-1

Update pt-br.html

EvandroZanatta commented:
would not be necessary to update the download links?
on the official website, the link to download the linux version is broken

remil19 committed:
Add license informations

remil19 commented:
There has been a lot of modifications in index.html the last twelve days but if we update the page on my repository it will still have to be updated every time a modification is made is the official repository so i think we should only update index.php when the merge will be done.

remil19 commented:
There is also some issues specific to the translations such as the one @cobaimelan raised (it happens also for Greek), another is some sentences are longer once translated so their end is hide by an image put next to it. I cannot make modifications impacting the design of the site, the official team have to make them (which font use, should we move the image or break the lines which are too long...)

It' s actually up to the official team now.

EvandroZanatta commented:
it should have been fused to the main repository.
